docs/architecture/ADR-003-Non-interactive-defaults.md

```diff
@@ -0,0 +1,698 @@
+
+The exact approach taken by the initial implementation, however, is certainly up for scrutiny.
+
+## Intro
+
+pls see the [original specs](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md), from which this ADR paraphrases heavily.
+
+Currently, when checking for message inclusion, validators recreate the share commitment from the messages in the block and compare those with what are signed over in the `MsgPayForData` transactions also in that block. If any commitment is not found in one of the PFD transactions, or if there is a commitment that doesn't have a corresponding message, then they reject that block.
+
+While this functions as a message inclusion check, the light client has to assume that 2/3's of the voting power is honest in order to be assured that both the messages they are interested in and the rest of the messages paid for in that block are actually included. In order to have this property, we need a block validity rule where:
+
+> **All share commitments included in `MsgPayForData` must consist only of subtree roots of the data square.**
+
+The main issue with that requirement is that users must know the relevant subtree roots before they sign, which is problematic considering that if the block is not organized perfectly, the subtree roots will include data unknown to the user at the time of signing.
+
+To fix this, the spec outlines the “non-interactive default rules”. These involve a few additional **default but optional** message layout rules that enables the user to follow the above block validity rule, while also not interacting with a block producer. Commitments to messages can consist entirely of sub tree roots of the data hash, and for those sub tree roots to be generated only from the message itself (so that the user can sign something “non-interactively”).
+
+NOTE: THE STATEMENT BELOW IS MODIFIED FROM THE SPEC
+
+> **Messages must begin at a location aligned with the largest power of 2 that is not larger than the message length or k.**
+
+Below illustrates how we can break a message up into two different subtree roots, following a [Merkle Mountain Range](https://docs.grin.mw/wiki/chain-state/merkle-mountain-range/) structure, i.e. root `A` for first four shares, and root `B` consisting of the last two shares. We can then create a commitment out of any number of subtree roots by creating a merkle root of those commitments.
+
+![Subtree root commitment](./assets/subtree-root.png "Subtree Root based commitments")
+
+> **If the messages are larger than k, then they must start on a new row.**
+
+![Multi row share commitment](./assets/multirow.png "Subtree Root based commitments")
+
+If we can follow these rules, we can always create a commitment to the data that is a subtree root of the data root while only knowing the data in that message.
```

Comment:
Hmmm not sure this is worded correctly. It should be something like "a commitment to subtree roots of the data root, that commit to the data"?

Reply:
good point e8842c8457397e36bd74e78230c5c79724ed89db